Monika After Story/game/script-affection.rpy

```diff
@@ -173,21 +173,19 @@ init python:
 
     #Monika's initial affection based on start-up.
     if persistent._mas_long_absence == False:
+        $ current_time = datetime.now()
         if persistent.sessions["last_session_end"] == None:
             pass
-        elif datetime.datetime.now() >= persistent.sessions["last_session_end"] + datetime.timedelta(hours = 6) and datetime.datetime.now() <= persistent.sessions["last_session_end"] + datetime.timedelta(hours = 12):
+        elif current_time >= persistent.sessions["last_session_end"] + datetime.timedelta(hours = 6) and current_time <= persistent.sessions["last_session_end"] + datetime.timedelta(hours = 12):
```

Comment:
Just noticed that you compare datetimes to current times by calculating the specific start/end points and comparing them to the current time. Subtracting two datetimes will give you a timedelta, so it's better to just subtract the last session time from the current time and compare it to the appropriate timedelta:
```python
current_time - persistent.sessions["last..."] >= datetime.timedelta(hours=6)
```
And to handle checking within a specific range, start your conditional checking with the largest value:
```python
if more than 4 weeks:
   <code>
elif more than 3 weeks:
   <code>
elif more than 2 weeks:
...
```
This allows you to not have to check both start and end values for a range, since the upper limits will be handled by the conditionals prior to the statements.